tensorflow_quantum/core/ops/noise/noisy_samples_op_test.py

```diff
@@ -0,0 +1,289 @@
+
+    @parameterized.parameters([{
+        'channel': x
+    } for x in util.get_supported_channels()])
+    def test_single_channel(self, channel):
+        """Individually test adding just a single channel type to circuits."""
+        symbol_names = []
+        batch_size = 3
+        n_qubits = 5
+        qubits = cirq.GridQubit.rect(1, n_qubits)
+
+        circuit_batch, resolver_batch = \
+            util.random_circuit_resolver_batch(
+                qubits, batch_size, include_channels=False)
+
+        for i in range(batch_size):
+            circuit_batch[i] = circuit_batch[i] + channel.on_each(*qubits)
+
+        symbol_values_array = np.array(
+            [[resolver[symbol]
+              for symbol in symbol_names]
+             for resolver in resolver_batch])
+
+        n_samples = (2**n_qubits) * 1000
+
+        op_samples = noisy_samples_op.samples(
+            util.convert_to_tensor(circuit_batch), symbol_names,
+            symbol_values_array, [n_samples]).to_list()
+        op_hists = self._compute_hists(op_samples, n_qubits)
+
+        cirq_samples = batch_util.batch_sample(circuit_batch, resolver_batch,
+                                               n_samples,
+                                               cirq.DensityMatrixSimulator())
+        cirq_hists = self._compute_hists(cirq_samples, n_qubits)
+
+        for a, b in zip(op_hists, cirq_hists):
+            self.assertLess(stats.entropy(a + 1e-8, b + 1e-8), 0.15)
+
+    def test_correct_padding(self):
+        """Test the variable sized circuits are properly padded."""
+        symbol_names = []
+        batch_size = 2
+        n_qubits = 5
+        qubits1 = cirq.GridQubit.rect(1, n_qubits)
+        qubits2 = cirq.GridQubit.rect(1, n_qubits + 1)
+
+        circuit_batch1, resolver_batch = \
+            util.random_circuit_resolver_batch(
+                qubits1, batch_size, include_channels=True)
+
+        circuit_batch2, _ = \
+            util.random_circuit_resolver_batch(
+                qubits2, batch_size, include_channels=True)
+
+        symbol_values_array = np.array(
+            [[resolver[symbol]
+              for symbol in symbol_names]
+             for resolver in resolver_batch] * 2)
+
+        n_samples = 10
+
+        op_samples = noisy_samples_op.samples(
+            util.convert_to_tensor(circuit_batch1 + circuit_batch2),
+            symbol_names, symbol_values_array, [n_samples]).to_list()
+        a_reps = np.asarray(op_samples[:2])
+        b_reps = np.asarray(op_samples[2:])
+        self.assertEqual(a_reps.shape, (2, 10, 5))
+        self.assertEqual(b_reps.shape, (2, 10, 6))
+
+    def test_correctness_empty(self):
+        """Test the expectation for empty circuits."""
+        empty_circuit = util.convert_to_tensor([cirq.Circuit()])
+        empty_symbols = tf.convert_to_tensor([], dtype=tf.dtypes.string)
+        empty_values = tf.convert_to_tensor([[]])
+        empty_n_samples = tf.convert_to_tensor([1], dtype=tf.int32)
```

Comment:
The `num_samples` parameter doesn't have a `batch_size` first dimension so regardless of batch_size this will always need to have size 1 for valid inputs. So in this input batch_size is zero meaning empty, but we do need to keep one entry in this tensor.